src/nvim/diff.c

```diff
@@ -1074,6 +1092,69 @@ static int diff_file_internal(diffio_T *diffio)
   memset(&emit_cfg, 0, sizeof(emit_cfg));
   memset(&emit_cb, 0, sizeof(emit_cb));
 
+  if (diffexpr_internal_prefix()) {
```

Comment:
Okay, I'll refactor this. This is (yet) another consequence of me not being comfortable with `call_func_retlist` and having fallen back to `eval_to_bool` instead.

Reply:
Hm, I cannot do *everything* within `eval_diff_fn` at least without messing with include headers, because it seems that `diffio_T` or `xdiff_out` are not available within `eval.c`. I'll move this "translation" process to a dedicated function, then call it at least from `diff_file` for now.